cmd/src/server.rs

```diff
@@ -853,10 +882,19 @@ impl<ER: RaftEngine> TiKVServer<ER> {
     }
 
     fn init_metrics_flusher(&mut self, fetcher: BytesFetcher) {
-        self.background_worker.start_with_timer(
-            "metrics-flusher",
-            MetricsFlusher::new(self.engines.as_ref().unwrap().engines.clone(), fetcher),
-        );
+        let remote = self.background_worker.clone_yatp_handle();
+        let mut interval = GLOBAL_TIMER_HANDLE
+            .interval(Instant::now(), DEFAULT_METRICS_FLUSH_INTERVAL)
+            .compat();
+        let mut engine_task =
+            EngineMetricsTask::new(self.engines.as_ref().unwrap().engines.clone());
+        let mut io_task = IOMetricsTask::new(fetcher);
+        remote.spawn(async move {
+            while let Some(Ok(_)) = interval.next().await {
```

Comment:
how can it stop?

Reply:
I am not sure. Can yatp stop itself when there are pending tasks? If it's a problem we can use a channel like before.

Reply:
Seems to me that `Worker` itself doesn't stop the channel either.